src/libutil/hash.cc

```diff
@@ -16,6 +17,28 @@
 
 namespace nix {
 
+#if OPENSSL_VERSION_NUMBER < 0x10101000L
```

Comment:
I think we can get rid of this now, since this is a very old version of OpenSSL.

Reply:
OpenSSL <1.1.1 removed, configure check added.